### DISCUSSION TITLE
I need to add button import CSV file in app

### DISCUSSION THREAD

**Discussion by tetswww** [Budibase/budibase#12306] [Help] [answered]
**Describe the feature request**
I need to add button import CSV file in app of list page 
import file like show all field with data and import with relation and all condition 

Please give me suggestion for same if any possibility in budibase

**Comment by ConorWebb96** [comment]
Hey @tetswww,

You can import CSVs into the data area by clicking the `create new table` button. Then clicking the upload button and selecting the CSV you wish to import. This won't import the relationships, as we can't be sure that the relationships will exist on our end therefore it is ignored.

<img width="1512" alt="Screenshot 2023-11-08 at 07 51 17" src="https://github.com/Budibase/budibase/assets/126772285/7d429718-8efc-4fb7-94a2-622e764353aa">
<img width="417" alt="Screenshot 2023-11-08 at 07 51 36" src="https://github.com/Budibase/budibase/assets/126772285/c5a4e20d-9035-4bc7-98d0-71c80ca007bd">

Additionally you are able to use a CSV as the basis for a data provider within your design area.

You can do this with various components such as: data provider, table block, or grid.  Changing the data option from a table to the bottom option `JSON/CSV`. 

You should be able to see a cog show up beside this selectable, when this cog is clicked a drawer should open providing you with a few different options. One being the ability to load a CSV file to use as the source for the data provider. Doing this will allow you to select a CSV file to use.

<img width="1511" alt="Screenshot 2023-11-08 at 07 46 33" src="https://github.com/Budibase/budibase/assets/126772285/5057faa8-6bf1-46b9-8a1b-d06d6cc75ee2">
<img width="313" alt="Screenshot 2023-11-08 at 07 47 04" src="https://github.com/Budibase/budibase/assets/126772285/41f32f58-efc6-479b-bda0-b0b86d4d6072">
<img width="888" alt="Screenshot 2023-11-08 at 07 47 30" src="https://github.com/Budibase/budibase/assets/126772285/2606928e-3415-4c3a-be1b-643d0358eadb">

Hopefully this helps.

**Reply by tetswww** [reply]
Hello ConorWebb96
I have not seen any option like csv json in list 
![git add pic](https://github.com/Budibase/budibase/assets/149157189/664ce7c5-9709-4df8-9bec-3c1b26b32b6a)
   
   Please give me suggestion for same how i can do this ?

**Reply by ConorWebb96** [reply]
What version of Budibase are you using?

**Reply by tetswww** [reply]
![image](https://github.com/Budibase/budibase/assets/149157189/77ce2bde-6b94-4c02-87fb-327698780340)

**Reply by ConorWebb96** [reply, marked as answer]
Yep you're 5 minor versions behind, there have been a lot of updates since version 2.8.10. I'd recommend updating Budibase to version 2.13 as you will get all the new features. Including the one shown above.

**Reply by tetswww** [reply]
Thank you 
can you please give me any suggestion how to upgrade version  without effect current app which is I was created now.
